detection_rules/rule.py

```diff
@@ -966,18 +983,30 @@ def get_version_space(self) -> Optional[int]:
 
     @property
     def latest_version(self) -> Optional[int]:
-        """Retrieve the latest known version of the rule."""
-        min_stack = self.get_supported_version()
-        return self.version_lock.get_locked_version(self.id, min_stack)
+        """Retrieve the latest known version of the rule, considering the version lock setting."""
+        toml_version = self.data.get("version")
+
+        if BYPASS_VERSION_LOCK and toml_version:
+            return toml_version
+
+        if toml_version:
```

Comment:
This is important, nice job

A thought - should we make this slightly stricter and add to the validation criteria that if `not BYPASS_VERSION_LOCK` then `version` and `revision` are invalid fields?